pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -238,135 +239,846 @@ func TestReconcile(t *testing.T) {
 		PipelineResources: []*v1alpha1.PipelineResource{gitResource, anotherGitResource, imageResource},
 	}
 	for _, tc := range []struct {
-		name          string
-		taskRun       *v1alpha1.TaskRun
-		wantBuildSpec buildv1alpha1.BuildSpec
+		name                   string
+		taskRun                *v1alpha1.TaskRun
+		wantServiceAccountName string
+		wantPodVolume          []corev1.Volume
+		wantPreSteps           []corev1.Container
+		wantSteps              []corev1.Container
```

Comment:
oh man you got through it! thanks so much for fixing this up @pivotal-nader-ziada :heart: :heart: :heart: i think this is definitely better going forward

Reply:
Yeah, this is great! I have given you quite a few merge conflicts though because of #488, sorry! I would lean towards using a single `wantPod` field instead of putting all of the individual fields so that we can match on anything in `ObjectMeta` or any additional fields as needed in the future as well.

Reply:
haha, that one file took more time than the rest of the PR :D

Reply:
> I would lean towards using a single `wantPod` field instead of using all of the individual fields so that we can match on anything in `ObjectMeta` or any additional fields as needed in the future as well.

The only "trouble" here @dwnusbaum is that there ends up being a lot of repetition between the pods so the cases become pretty verbose :S (maybe it's worth it tho?)

Side note: might start looking a bit better if we add pod builders in `test/builder` but that's probably a PR in itself :D

Reply:
> might start looking a bit better if we add pod builders in `test/builder`

Yeah that was what I was thinking, but like you said probably best left for another PR since the current code gets the job done.